ignite/metrics/negative_log_likelihood.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import division
+
+from torch.autograd import Variable
+from torch.nn.functional import nll_loss
+
+from .metric import Metric
+from ignite.exceptions import NotComputableError
+
+
+class NegativeLogLikelihood(Metric):
+    def __init__(self):
+        self._sum = 0
+        self._num_examples = 0
+
+    def reset(self):
+        self._sum = 0
+        self._num_examples = 0
+
+    def update(self, output):
+        y_pred, y = output
+        self._sum += nll_loss(Variable(y_pred), Variable(y)).data[0]
+        self._num_examples += 1
```

Comment:
I think this will end up giving the wrong results, as one usually use a batch size > 1 for training / testing. One way to address this is to pass a `num_samples` argument to `update`, or to try to infer the batch size from the data (first dimension?). Inferring from the first dimension is error-prone though, as there are some cases (NLP?) where the batch is not the first dimension I think.

Reply:
Yup, you're right - good spot. I've done this instead:
```
loss = nll_loss(Variable(y_pred), Variable(y), reduce=False).data
```

Reply:
This works as well, but I think `nll_loss(..., size_average=False)` is slightly more efficient.

Reply:
but then we don't get the size out, just the sum

Reply:
The size can be obtained from `y`, right?

Reply:
Is that error prone for NLP applications as you mentioned above?

Reply:
well, I just checked what ClassNLLCriterion and it [uses the first dim](https://github.com/pytorch/pytorch/blob/6279367297287615ed4e45bb5f117f1e11bfddd9/aten/src/THNN/generic/ClassNLLCriterion.c#L107), so I'll also update to this then